### DISCUSSION TITLE
Deno language

### DISCUSSION THREAD

**Discussion by mnemosdev** [denoland/deno#9706] [Q&A]
Hello, I would like to dive into the Deno language and how it was created. Is there anyone that could point me at how to do that?

I have tried to read the code but I don't know where to start and the logic behind it as I've never engineered a new language.

Thanks to anyone who might answer my call for knowledge!

**Comment by kitsonk** [comment]
Deno isn't a new language. Deno is a runtime for JavaScript, TypeScript and WebAssembly.

**Reply by mnemosdev** [reply]
Yes, as far as theory is concerned I can follow along, but I'd like to know where and how to start to delve in the inner workings of Deno.

Thanks!

**Reply by RajendrasinhParmar** [reply]
I have also just started to go through it. Gone through documents in the docs folder.
However, I would also like to know more and understand how to go through the whole repository and where to start.

**Reply by Foxhunt** [reply]
Hi @mnemosdev and @RajendrasinhParmar I'm also new to deno. I am currently working my way through the manual and found some info on the [architecture](https://deno.land/manual@v1.8.0/contributing/architecture) maybe you want to start there.

**Reply by RajendrasinhParmar** [reply]
Thank you for the response. I've already gone through those documents. And that seems very useful for the understanding of the architecture of deno. Along with that I would like to start contributing to the project. Can you provide more details on how exactly I can start doing contributions?
